### ISSUE TITLE
Create component => src/components/SitemapDiagram.tsx

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#649]
Create a React component for src/components/SitemapDiagram.tsx. Here are the instructions:

### Overview
We need to create a new React component called `SitemapDiagram.tsx` that will display a sitemap diagram for a given project. This component will be used on the project sitemap page to help users visualize the structure of their project's files. The component should be written in TypeScript using ES6 syntax, such as arrow functions, and styled using Tailwind CSS.

### Requirements

1. Create a new file `SitemapDiagram.tsx` inside the `src/components` folder.
2. Import the necessary dependencies at the top of the file:

```typescript
import React from 'react';
import { Sitemap } from '~/types';
```

3. Define a `Props` interface for this component that includes a `sitemap` prop, which is an array of `Sitemap` objects from the `types.ts` file:

```typescript
interface Props {
  sitemap: Sitemap[];
}
```

4. Create a functional component `SitemapDiagram` that takes the `Props` interface as its type:

```typescript
const SitemapDiagram: React.FC<Props> = ({ sitemap }) => {
  // Component logic and rendering
};
```

5. Inside the component, map over the `sitemap` prop and create a visual representation of each file in the sitemap. Use Tailwind CSS classes for styling and layout. For example, you can use `flex`, `flex-col`, `bg-gray-100`, `p-4`, `rounded`, and other classes to style the sitemap diagram.

```typescript
const renderSitemapItems = () => {
  return sitemap.map((item) => (
    <div key={item.id} className="bg-gray-100 p-4 rounded mb-4">
      <h3 className="font-bold text-lg mb-2">{item.fileName}</h3>
      <p className="text-gray-600">{item.fileDescription}</p>
    </div>
  ));
};
```

6. In the component's return statement, render a container `div` that wraps the `renderSitemapItems()` function call:

```typescript
return (
  <div className="flex flex-wrap gap-4">
    {renderSitemapItems()}
  </div>
);
```

7. Export the `SitemapDiagram` component as the default export:

```typescript
export default SitemapDiagram;
```

### Summary
To recap, the `SitemapDiagram.tsx` component should:

- Be a functional React component written in TypeScript using ES6 syntax.
- Accept a `sitemap` prop, which is an array of `Sitemap` objects from the `types.ts` file.
- Render a visual representation of each file in the sitemap using Tailwind CSS classes.
- Export the `SitemapDiagram` component as the default export.

Once the component is created, please submit a pull request for review. If you have any questions or need clarification, feel free to reach out.
